app/grandchallenge/components/views.py

```diff
@@ -84,3 +89,46 @@ def get_queryset(self):
 
     def get_result_label(self, result):
         return result.title
+
+
+class InterfaceProcessingMixin:
+    def process_data_for_object(self, data):
+        raise NotImplementedError
+
+    def form_invalid(self, form):
+        return JsonResponse(form.errors, status=400)

```

Comment:
Missing `super()`?

Reply:
In this case, super().form_invalid() only renders the form with error messages, right?. We can call it, but we wouldn't do anything with the output since we need to return the errors in json format for our custom js code to pick them up and add them to the form correctly.

Reply:
That may be the case now but you do not know that in the future for all uses of this Mixin. It is best practise to always call the super method as the view could be doing something custom.